NDTensors/test/test_diag.jl

```diff
@@ -31,22 +31,30 @@ using LinearAlgebra: dot
     @test complex(D) == Diag(one(complex(elt)))
     @test similar(D) == Diag(0.0)
 
-    D = Tensor(Diag(1), (2, 2))
+    D = dev(Tensor(Diag(1), (2, 2)))
     @test norm(D) == √2
     d = 3
     vr = rand(elt, d)
     D = dev(tensor(Diag(vr), (d, d)))
     Da = Array(D)
     Dm = Matrix(D)
+    Da = permutedims(D, (2, 1))
     @allowscalar begin
       @test Da == NDTensors.LinearAlgebra.diagm(0 => vr)
       @test Da == NDTensors.LinearAlgebra.diagm(0 => vr)
 
-      ## TODO Currently this permutedims requires scalar indexing on GPU.
-      Da = permutedims(D, (2, 1))
       @test Da == D
     end
 
+    if (dev == NDTensors.mtl && elt != ComplexF32)
```

Comment:
I reported a bug in metal that causes this code to fail.
https://github.com/JuliaGPU/Metal.jl/issues/364

Reply:
Could you add a link to that issue as a comment in the code?

Reply:
Yes I added that comment. Thanks!

Reply:
The logic of this if-statement doesn't seem right to me. Are you sure this will run on CPU and also on other GPU backends?

Reply:
From what I can tell it will only run on Metal if the element type is `Float32`, but not on other device backends.

Reply:
Maybe it should be:
```julia
if !(dev == NDTensors.mtl && elt == ComplexF32)
  # ...
end
```

Reply:
Oh good catch thank you for fixing that logic. I have only been testing on my mac and didn't catch that. Fixed now!